homeassistant/components/nightscout/sensor.py

```diff
@@ -28,10 +45,37 @@ async def async_setup_entry(
     entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the Glucose Sensor."""
+    """Set up entities."""
     api = hass.data[DOMAIN][entry.entry_id]
+
+    # Glucose sensor
     async_add_entities([NightscoutSensor(api, "Blood Sugar", entry.unique_id)], True)
 
+    # Uploader batteries
+    async def async_update_batteries():
+        """Fetch the latest data from Nightscout REST API and update the state of devices batteries."""
+        try:
+            return await api.get_latest_devices_status()
+        except OSError as error:
+            _LOGGER.error(
+                "Error fetching battery devices status. Failed with %s", error
+            )
+            raise UpdateFailed(f"Error communicating with API: {error}") from error
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="battery_sensor",
+        update_method=async_update_batteries,
+        update_interval=timedelta(minutes=1),
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+    async_add_entities(
```

Comment:
Collect all entities of the platform first and then make a single call to add entities.